src/raftstore/store/snap.rs

```diff
@@ -383,14 +732,36 @@ impl SnapManagerCore {
         self.registry.contains_key(key)
     }
 
-    #[inline]
-    pub fn get_snap_file(&self, key: &SnapKey, is_sending: bool) -> io::Result<SnapFile> {
-        SnapFile::new(&self.base, self.snap_size.clone(), is_sending, key)
+    pub fn get_send_snapshot_file_creator(&self,
+                                          key: &SnapKey)
+                                          -> io::Result<Box<SendSnapshotFileCreator>> {
+        let f = try!(v1::SnapFile::new(&self.base, self.snap_size.clone(), true, key));
+        Ok(Box::new(f))
+    }
+
+    pub fn get_send_snapshot_file(&self, key: &SnapKey) -> io::Result<Box<SendSnapshotFile>> {
+        let reader = try!(v1::SendSnapFile::new(&self.base, self.snap_size.clone(), key));
+        Ok(Box::new(reader))
+    }
+
+    pub fn get_recv_snapshot_file_writer(&self,
+                                         key: &SnapKey,
+                                         _data: &[u8])
```

Comment:
It's preserved for the new format. These fields in `RaftSnapshotData` are used to encoded cf file sizes in new format.